pkg/k8s/apis/cilium.io/v2/bgp_cluster_types.go

```diff
@@ -126,13 +126,39 @@ type CiliumBGPPeer struct {
 	// +kubebuilder:default=0
 	PeerASN *int64 `json:"peerASN,omitempty"`
 
+	// AutoDiscovery is the configuration for auto-discovery of the peer address.
+	//
+	// +kubebuilder:validation:Optional
+	AutoDiscovery *AutoDiscovery `json:"autoDiscovery,omitempty"`
+
 	// PeerConfigRef is a reference to a peer configuration resource.
 	// If not specified, the default BGP configuration is used for this peer.
 	//
 	// +kubebuilder:validation:Optional
 	PeerConfigRef *PeerConfigReference `json:"peerConfigRef,omitempty"`
 }
 
+// AutoDiscovery is the configuration for auto-discovery of the peer address.
+type AutoDiscovery struct {
```

Comment:
Could you prefix this type name (as well as DefaultGateway) with `BGP`? -> `BGPAutoDiscovery`

(This package is common for all cilium APIs, to make it clear that it belongs to the BGP feature)

Reply:
makes sense. fixed